src/driver/argument_parser.hpp

```diff
@@ -590,6 +590,12 @@ struct argument_parser
                 v(*this, inputs);
             return arg.action(*this, inputs);
         }
+        catch(const std::runtime_error& e)
+        {
+            msg = e.what();
+            std::cerr << "error: " << msg << std::endl;
+            return false;
+        }
```

Comment:
This is wrong as this will bypass all the spellcheking. For example, running `./bin/driver read -h2` should output:

```
error: Found argument '-h2' which wasn't expected, or isn't valid in this context
       Did you mean -h?

USAGE:
    ./bin/driver read -h
```

However, with this change it would skip the spellchecking and then continue to try to run the command(because you are returning `false`), So it would have this output:

```
error: Path does not exists: -h2
Reading:
terminate called after throwing an instance of 'std::length_error'
  what():  cannot create std::vector larger than max_size()
Aborted (core dumped)
```

Also, the `runtime_error` exception is already caught by `catch(const std::exception& e)` below, so we aren't ignoring the exception.

Reply:
@pfultz2 thanks. I don't understand the flow here, so please do explain what you find wrong. But in my mind, if a file isn't found, and an exception is raised it is getting ignored in the (prior) exception catchers.. 
Do we still want a spellcheck when a runtime exception has been raised..?

Reply:
> But in my mind, if a file isn't found, and an exception is raised it is getting ignored in the (prior) exception catchers..

I am not sure I follow, there is only one catch handler. What flags are you passing to the driver? And what is the expected output you would like?

>  Do we still want a spellcheck when a runtime exception has been raised..?

Yes. Otherwise we can't spellcheck incorrect arguments, like I showed in the example above.

Reply:
@pfultz2 PTAL at https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/1809 it has all the details of the exact command that I typed. I can also paste it here..

"And what is the expected output you would like?". No, I am not expecting anything. I just don't want to break anything. 

This code that I changed is just to show the 3 error cases. I don't mean it to be final, but the exception handling has to change a little bit..? I need clarity on what is to be done here.. Thanks.

Reply:
>  don't mean it to be final, but the exception handling has to change a little bit..?

The exception handling shouldn't need to be changed.